driver/src/models/mod.rs

```diff
@@ -0,0 +1,47 @@
+pub mod state;
+pub mod flux;
+pub mod order;
+
+pub use crate::models::state::State;
+pub use crate::models::flux::PendingFlux;
+pub use crate::models::order::Order;
+
+use sha2::{Digest, Sha256};
+use web3::types::H256;
+
+pub const TOKENS: u8 = 30;
+pub const DB_NAME: &str = "dfusion2";
+
+pub trait RollingHashable {
+    fn rolling_hash(&self) -> H256;
+}
+
+pub trait RootHashable {
+    fn root_hash(&self, valid_items: &[bool]) -> H256;
+}
+
+pub trait Serializable {
+    fn bytes(&self) -> Vec<u8>;
+}
+
+fn merkleize(leafs: Vec<Vec<u8>>) -> H256 {
```

Comment:
How does anyone ever use this if it isn't public? What am I missing here?

Reply:
It's a convenience method to help implement the RootHashable trait, which is public. External components should use the rootHashable trait and not this method directly.